pkg/config/release.go

```diff
@@ -135,3 +143,16 @@ func GetAllConfigsFromSHA(releaseRepoPath, sha string, logger *logrus.Entry) (*R
 
 	return config, nil
 }
+
+func GetChangedClusterProfiles(path, baseSHA string) (ret []ClusterProfile, err error) {
```

Comment:
For the other values we load and diff the deserialized values -- pros/cons for doing this differently?

Reply:
Pros:

- We don't need the actual contents of the files until we create the ConfigMaps.
- Even then, we only need the contents of the profiles that changed (none or one, 99% of the time).
- No need to `checkout` one version, then `checkout` back again (if we weren't already doing it, of course).
- No need to compare all files manually.
- No need to invent some ad-hoc hashing algorithm, which has to handle multiple files.

Cons:

- There was one minor issue with the `git` version we use in CI.
- Uses a semi-obscure, low-level `git` command.
- ?

Essentially, the problem is: which files changed from one revision to the other, and I think `git` is the appropriate tool for that. The fact that it already stores a hash of the contents of the directories is a very nice bonus.  We replace a lot of complicated steps with an algorithm that is essentially `exec+scanf`.

Plus, `git` is awesome =)

Reply:
Another con is we end up finding changes in different ways in one code-base. @petr-muller you have any objections to that?

Reply:
Cluster profiles have some unique characteristics that make this approach more beneficial. Not all of those items are true for the other cases, e.g. the first is not true for `ci-operator` configuration files, the last, for templates (although we could probably use this as a first pass in those cases as well).

The latest version of the code is nine lines, 431 characters long. Looking at the code we have for the other cases (and keeping in mind cluster profiles would require special treatment), it seems to be worth having a separate loading mechanism.

Reply:
I was thinking about this when I reviewed the PR, and at first I was thinking, "why do we need a second path doing the same", but ended thinking what Bruno did is really better, and the duplicate way is a reasonable price. I think we could migrate detecting changes in tempaltes to this way too, but I think that could be a tech debt-paying followup.